### PR TITLE
layers: Add 09200 09201 09202

### PR DESCRIPTION
**Pull request by arno-lunarg**
Also update related error messages

### CODE REVIEW

File: layers/core_checks/cc_device_memory.cpp [hunk 1]
```diff
@@ -618,17 +618,48 @@ bool CoreChecks::ValidateBindBufferMemory(VkBuffer buffer, VkDeviceMemory mem, V
...
-                                                 : "VUID-vkBindBufferMemory-descriptorBufferCaptureReplay-08112";
-            const LogObjectList objlist(buffer, mem);
-            skip |=
-                LogError(objlist, vuid,
-                         "%s: If buffer was created with the VK_BUFFER_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_EXT bit set, "
-                         "memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT bit set.",
-                         api_name);
+        if (enabled_features.descriptor_buffer_features.descriptorBufferCaptureReplay) {
+            if (buffer_state->createInfo.flags & VK_BUFFER_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_EXT) {
+                if ((!chained_flags_struct || !(chained_flags_struct->flags & VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT))) {
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_device_memory.cpp:623]: we could just do a

`VkMemoryAllocateFlags flags = chained_flags_struct ? chained_flags_struct->flags : 0;`

that way we don't have to check each time for `!chained_flags_struct` and the error message if you use `string_VkMemoryAllocateFlags(0)` will print `VkMemoryAllocateFlags(0)` which I find it more useful then printing "None"

File: layers/core_checks/cc_device_memory.cpp [hunk 2]
```diff
@@ -618,17 +618,48 @@ bool CoreChecks::ValidateBindBufferMemory(VkBuffer buffer, VkDeviceMemory mem, V
...
+                    const char *vuid = bind_buffer_mem_2 ? "VUID-VkBindBufferMemoryInfo-descriptorBufferCaptureReplay-08112"
+                                                         : "VUID-vkBindBufferMemory-descriptorBufferCaptureReplay-08112";
+                    const LogObjectList objlist(buffer, mem);
+                    skip |= LogError(
+                        objlist, vuid,
+                        "%s: Buffer was created with the VK_BUFFER_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_EXT bit set, but"
+                        "bound memory was not allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT bit set. It was allocated "
+                        "with the following bit(s) set: %s.",
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_device_memory.cpp:631]: I think these could all be a little simpler to read if we did something like

```
"%s: Buffer was created with the VK_BUFFER_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_EXT bit set,"
"but the bound memory was allocated with %s and needs VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT.",
```

Comment (arno-lunarg) [reply layers/core_checks/cc_device_memory.cpp:631]: Yes, wording is better!

File: layers/core_checks/cc_device_memory.cpp [hunk 3]
```diff
@@ -617,18 +617,45 @@ bool CoreChecks::ValidateBindBufferMemory(VkBuffer buffer, VkDeviceMemory mem, V
                              "memory must have been allocated with the VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT bit set.",
                              api_name);
         }
+        const VkMemoryAllocateFlags memory_allocate_flags = chained_flags_struct ? chained_flags_struct->flags : 0;
+        if (buffer_state->createInfo.flags & VK_BUFFER_CREATE_DESCRIPTOR_BUFFER_CAPTURE_REPLAY_BIT_EXT) {
+            if ((!chained_flags_struct || !(chained_flags_struct->flags & VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT))) {
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_device_memory.cpp:622]:
```suggestion
            if (!(memory_allocate_flags & VK_MEMORY_ALLOCATE_DEVICE_ADDRESS_BIT)) {
```

**Review by spencer-lunarg — APPROVED** [review]

**Pull request merged**